### ISSUE TITLE
Using Prisma `NextJS` edge runtime leads to error

### ISSUE THREAD

**Issue by homanp** [prisma/prisma#18763]
## Problem

Currently, running `@prisma/client` in NextJS `edge runtime` is not supported.

## Suggested solution

Add support for using `@prisma/client` in `edge` runtimes

**Comment by adamlong5** [comment]
Is this still the case now that Vercel has launched their own edge-supporting Postgres client?

**Comment by noc2spam** [comment]
Still facing the same problem. I get `Error: PrismaClient is unable to be run in the browser.` when experimental-edge is enabled.

**Comment by jeffminsungkim** [comment]
> Still facing the same problem. I get `Error: PrismaClient is unable to be run in the browser.` when experimental-edge is enabled.

+1

**Comment by janpio** [comment]
Also reported on Twitter: https://twitter.com/ajones55555/status/1654701659187195904

**Comment by amlcodes** [comment]
verified not fixed data-proxy and 4.15.0-dev.13 from #19173

**Comment by altechzilla** [comment]
This is still happening in edge runtime.

**Comment by colinricardo** [comment]
why do we still get this error even when using 

`prisma generate --data-proxy` 

and 

`import { PrismaClient } from "@prisma/client/edge"`?

the console output from the generate says: 

![CleanShot 2023-06-18 at 17 46 56@2x](https://github.com/prisma/prisma/assets/21294951/fc4ca8f3-6963-4a34-b189-9113e47b1f91)

what am i missing, shouldn't this work in vercel edge runtime?

**Comment by johannbuscail** [comment]
Any update ?

**Comment by SamueleBarbiera** [comment]
Any update with the prisma 5v?

**Comment by Jolg42** [comment]
Not yet, we'll definitely let you know here when we have an update. Stay tuned 📻

**Comment by joselvelez** [comment]
I just ran into this issue as well in my project. Any updates?

**Comment by sharpsteelsoftware** [comment]
Excited for a response on getting prisma in edge runtime

**Comment by janpio** [comment]
Hey, we at Prisma have been working on the foundations for Edge Functions support, including Next.js edge runtime, for the past few weeks. Soon, we'll have something to share and test! If you are interested in helping us, please respond to this super short survey and leave your email address so we can get in touch with instructions: https://pris.ly/survey/driver-adapter-nextjs-edge-runtime Thanks.

**Comment by Re4GD** [comment]
Any updates on this? Everybody is going edge and cold starts are becoming unbearable. We love Prisma and the DX is unmatched but this issue is pushing us to use alternative edge compatible solutions. Can you guys give us a status update, or an ETA, or a roadmap?

Really don’t want to drop Prisma 😁

**Comment by janpio** [comment]
We are working on this _right now_, have an internal working prototype (so know it is possible), and expect to share a first early access version with people who filled the survey in the next weeks.

**Comment by janpio** [comment]
We invited the first few people, yes. We'll react to their feedback now, get fixes implemented, and then invite more people.

**Comment by Gul109** [comment]
Any improvements?

**Comment by adxxtya** [comment]
has it launched yet @janpio ? I'm completely out of touch rn

**Comment by janpio** [comment]
Early Access ongoing with quite a few users now, but expect a public Preview release soon.

**Comment by jonathanwilke** [comment]
Can't wait for this. Hope you will launch it soon, otherwise I need to migrate [supastarter.dev](https://supastarter.dev) to Drizzle to support edge functions. I really want to keep Prisma though!

**Comment by Khaogamermain01** [comment]
Any updates on this? @janpio

**Comment by janpio** [comment]
Yes, watch our for our next release - it should include some _very_ relevant things related to using Next.js via edge runtimes in edge function environments.

**Comment by mdcarter** [comment]
> Yes, watch our for our next release - it should include some _very_ relevant things related to using Next.js via edge runtimes in edge function environments.

What a way of building suspense ! 😆

**Comment by janpio** [comment]
Well, I think that is ok a few hours before a release 😆 

---

Hey everyone!

We just released [Prisma ORM version 5.11.0](https://github.com/prisma/prisma/releases/tag/5.11.0) which includes a preview feature for **Edge Functions support via [Cloudflare Workers and Pages](https://www.prisma.io/docs/orm/prisma-client/deployment/edge/deploy-to-cloudflare) and [Vercel Edge Functions and Middleware](https://www.prisma.io/docs/orm/prisma-client/deployment/edge/deploy-to-vercel)** in Prisma ORM 🥳 

- The release notes give a quick overview: https://github.com/prisma/prisma/releases/tag/5.11.0
- We also have extensive documentation: https://www.prisma.io/docs/orm/prisma-client/deployment/edge/overview
- Prisma ORM now supports drivers for Neon Database (and Vercel Postgres), PlanetScale and Turso via their serverless drivers on both Vercel and Cloudflare, and additionally PostgreSQL via `pg` on Cloudflare 💥 

Please give it a try, and let us know how it goes! If you encounter any problems, please create a [new bug report issue](https://github.com/prisma/prisma/issues/new/choose), or if the problem is driver adapter specific, use the feedback discussions for [`@prisma/adapter-neon`](https://github.com/prisma/prisma/discussions/21346), [`@prisma/adapter-planetscale`](https://github.com/prisma/prisma/discussions/21347), [`@prisma/adapter-libsql` / Turso](https://github.com/prisma/prisma/discussions/21345) or [`@prisma/adapter-pg`](https://github.com/prisma/prisma/discussions/22899) 🙇 

PS: We are also [working on support for Cloudflare D1](https://github.com/prisma/prisma/issues/13310), and expect to share news in this feature request issue soon. (Early Access for that is already in progress)

**Comment by Flip-Liquid** [comment]
awesome, great work on this @janpio. it sounds like `node-postgres` support on the Vercel edge is not supported on Vercel's side, is that correct? If not, is that something being considered or worked on?

**Comment by janpio** [comment]
Yes, that is correct. Vercel does not support using `node-postgres` in their edge runtime, so there is unfortunately nothing we can do about that. As soon as you can use `node-postgres` on Vercel Edge Functions and Middleware it will also work when using Prisma. (If you are using Vercel Postgres, you can use the Neon Serverless Driver though as a workaround to talk to your PostgreSQL database. Unfortunately only for Vercel Postgres and Neon itself though.)

**Comment by jkomyno** [comment]
Hi, Alberto from Prisma here.

Running Prisma with Next.js on Vercel Edge runtime [has been possible for some time](https://github.com/prisma/prisma/issues/18763#issuecomment-1992318263), but it got even better now!

You'll have to use the new [`prisma-client` generator](https://www.prisma.io/docs/orm/prisma-schema/overview/generators#prisma-client-preview), plus the `queryCompiler` and `driverAdapters` preview features (see: [Prisma ORM without Rust engines](https://www.prisma.io/docs/orm/prisma-client/setup-and-configuration/no-rust-engine)).

```
generator client {
  provider = "prisma-client"
  output = "../lib/generated/prisma"
  previewFeatures = ["driverAdapters", "queryCompiler"]
}
```

If you exclusively use Prisma on Vercel Edge, you should specify:

```diff
generator client {
  provider = "prisma-client"
+ runtime = "edge-light" // you can also use "vercel"
  // ...
}
```

otherwise, if you need to use both the Node.js + Vercel Edge runtimes, you can do it by specifying two different clients:

```diff
generator client {
  provider = "prisma-client"
  output = "../lib/generated/prisma"
+ runtime = "nodejs"
  // ...
}

generator edge {
  provider = "prisma-client"
  output = "../lib/generated/prisma-edge"
+ runtime = "edge-light"
  // ...
}
```

[Here](https://github.com/prisma/prisma-examples/tree/feat/prisma-client-fixes/generator-prisma-client/nextjs-starter-webpack-with-middleware)'s an official example using Prisma's latest dev version combining Vercel's usual Node.js runtime with Next.js middlewares running on Vercel Edge. We recommend waiting next week for Prisma 6.14.0 for your production applications.

Note that `pg` (and `@prisma/adapter-pg`) remain incompatible with Vercel Edge (see potentially alternative feature request: https://github.com/prisma/prisma/issues/23744).

Thanks!
